cwms-data-api/src/main/java/cwms/cda/api/watersupply/WaterContractUpdateController.java

```diff
@@ -0,0 +1,122 @@
+
+    private Timer.Context markAndTime(String subject) {
+        return Controllers.markAndTime(metrics, getClass().getName(), subject);
+    }
+
+    public WaterContractUpdateController(MetricRegistry metrics) {
+        this.metrics = metrics;
+    }
+
+    @NotNull
+    protected WaterContractDao getContractDao(DSLContext dsl) {
+        return new WaterContractDao(dsl);
+    }
+
+    @OpenApi(
+        requestBody = @OpenApiRequestBody(
+            content = {
+                @OpenApiContent(from = WaterUserContract.class, type = Formats.JSONV1)
+            },
+            required = true),
+        queryParams = {
+            @OpenApiParam(name = NAME, description = "Specifies the new name of the contract.", required = true)
+        },
+        pathParams = {
+            @OpenApiParam(name = CONTRACT_ID, description = "Specifies the name of the contract to be renamed.",
+                    required = true),
+            @OpenApiParam(name = OFFICE, description = "The office Id the contract is associated with.",
+                    required = true),
+            @OpenApiParam(name = PROJECT_ID, description = "The project Id the contract is associated with.",
+                    required = true),
+            @OpenApiParam(name = WATER_USER, description = "The water user the contract is associated with.",
+                    required = true)
+        },
+        responses = {
+            @OpenApiResponse(status = "404", description = "The provided combination of "
+                    + "parameters did not find a contract"),
+            @OpenApiResponse(status = "501", description = "Requested format is not implemented.")
+        },
+        description = "Renames a water contract",
+        method = HttpMethod.PATCH,
+        path = "/projects/{office}/{project-id}/water-users/{water-user}/contracts/{contract-id}",
+        tags = {TAG}
+    )
+
+    @Override
+    public void handle(@NotNull Context ctx) {
+        try (Timer.Context ignored = markAndTime(UPDATE)) {
+            DSLContext dsl = getDslContext(ctx);
+            String contractName = ctx.pathParam(CONTRACT_ID);
+            String formatHeader = ctx.header(Header.ACCEPT) != null ? ctx.header(Header.ACCEPT) : Formats.JSONV1;
+            ContentType contentType = Formats.parseHeader(formatHeader, WaterUserContract.class);
+            ctx.contentType(contentType.toString());
+            String newName = ctx.queryParam(NAME);
+            WaterUserContract waterContract = Formats.parseContent(contentType, ctx.body(), WaterUserContract.class);
+            WaterContractDao contractDao = getContractDao(dsl);
+            WaterUser ref = new WaterUser.Builder().withEntityName(waterContract.getWaterUser().getEntityName())
+                    .withProjectId(waterContract.getWaterUser().getProjectId())
+                    .withWaterRight(waterContract.getWaterUser().getWaterRight()).build();
+            contractDao.renameWaterContract(ref, contractName, newName);
```

Comment:
why does this need a full WaterUser instead of just id's?

Reply:
Package call in DAO requires full wateruser object

Reply:
strange that it needs so much...